src/target_clones.rs

```diff
@@ -63,34 +88,21 @@ impl ToTokens for FunctionClone<'_> {
                 }
             })
             .collect::<Vec<_>>();
-        // We create 3 different functions here: outer, inner, recursion_helper.
-        // The outer function is the main one and in principle do not need
-        // anything more than that one. The other two functions solve specific
-        // problems.
-        // The reason for the function inner is that we have to mark the outer
-        // funtion as unsafe to be able to use target_feature, however to do not
-        // not necessarily want the body to be unsafe. The solution is to place
-        // it in an inner function, mark that function as inline(always) and
-        // only ever call it from a single location.
-        // The only remaining problem is now that of recursion. We would like
-        // recursion to work as expected, without breaking the optimizations
-        // above. We would also like it to call this function without going
-        // through the dispatcher. The solution is to create a function in
-        // scope with the same name and signature. This function will simply
-        // call ourselves.
+
+        // We create an inner and an outer function here.
+        //
+        // The outer function is invoked by the dispatcher. It must be marked unsafe since it uses
+        // the `target_feature` attribute.
+        //
+        // The inner function contains the actual implementation of the function. This solves two
+        // problems. First, this function has the same safety as the outermost function, allowing
+        // normal safety guarantees within the unsafe outer function. Second, this function has the
+        // same name as the outermost function, allowing recursion to skip nested feature
+        // detection.
         tokens.extend(quote! {
             #target_arch
             #target_features
-            #outer_signature {
-                #target_arch
-                #[inline(always)]
-                #recursion_helper_signature {
-                    unsafe {
-                        #outer_function_ident(#(#argument_names),*)
-                    }
-                }
-
-                #target_arch
+            pub #outer_signature {
```

Comment:
I think this should probably be `max(outer_function_publicness, "pub(super)")`

Reply:
I'm not sure it matters though.

Reply:
I think that would prevent static dispatching across modules, right?  I'm not sure that's desirable.  I'm not even opposed to someone making the generated functions crate-public (though I wouldn't do it)